dev branch

abocquierCC opened the pull request.

abocquierCC commented on .config/Dockerfile:
```diff
@@ -1,7 +1,7 @@
 ARG grafana_image=grafana-enterprise
 
-FROM grafana/${grafana_image}:${grafana_version}
+FROM grafana/${grafana_image}:11.5.0
```
ok, but we need to have a method to manage grafana version (for example a config file to update the version of grafana v11 to v12)

abocquierCC commented on .github/workflows/playwright.yml:
```diff
@@ -0,0 +1,59 @@
+          done
+          echo "Grafana did not become ready in time."
+          docker compose -f docker-compose-new.yaml logs
+          exit 1
```
ok, 
can't we use a docker feature to wait the container ?

add comment to add detail on timing

zahed942001 replied:
done:
 STATUS=$(docker inspect --format='{{.State.Health.Status}}' clevercloud-warp10-datasource)

abocquierCC commented on .github/workflows/playwright.yml:
```diff
@@ -0,0 +1,59 @@
+        with:
+          name: playwright-report
+          path: playwright-report/
+          retention-days: 30
```
I think you can reduce the retention option, we don't need 30 days

zahed942001 replied:
Done: 
Changed retetion-days to 5

abocquierCC commented on .github/workflows/warp10-e2e-editor.yml:
```diff
@@ -0,0 +1,55 @@
+name: warp10-e2e-test
```
This file seems similar to the previous one (`playwright.yml`) any reason ?

zahed942001 replied:
Done:
removed

abocquierCC commented on dist/module.js:
in a near future we will not commit this folder, it will build by the CI. Then don't commit it and manage its build in the CI to use it

zahed942001 replied:
Done:
added to .gitinore

abocquierCC commented on docker-compose-new.yaml:
```diff
@@ -0,0 +1,43 @@
+    build:
+      context: ./.config
+      args:
+        grafana_version: ${GRAFANA_VERSION:-11.5.0}
```
> ok, but we need to have a method to manage grafana version (for example a config file to update the version of grafana v11 to v12)

we will need this to

abocquierCC commented on docker-compose-new.yaml:
move this file in a new directory to separate it from the previous docker-compose: `tests/config/`

zahed942001 replied:
Done:
New file named docker-compose-plugin

abocquierCC commented on package.json:
```diff
@@ -71,8 +72,9 @@
-    "@grafana/ui": "^11.3.2",
     "@grafana/schema": "^11.3.2",
+    "@grafana/ui": "^11.3.2",
+    "fs": "^0.0.1-security",
```
what this lib do ?

abocquierCC replied:
needed by the webpack, we should update the sdk

abocquierCC commented on playwright.config.ts:
```diff
@@ -4,48 +4,31 @@ import { dirname } from 'node:path';
       name: 'chromium',
-      use: { ...devices['Desktop Chrome'], storageState: 'playwright/.auth/admin.json' },
+      testDir: './tests',
+      testMatch: ['**/query-editor.spec.ts','**/constant-editor.spec.ts'],
```
you don't need to specify each test files `*.spec.ts`

zahed942001 replied:
Done:
      testMatch: ['*.spec.ts'],

abocquierCC commented on tests/constant-editor.spec.ts:
rename this tests file according to the scenario and place in a director that represent the scope of the test

here, for example: `datasource/test-datasource-config-componenent`

zahed942001 replied:
Renamed and seperated to multiple test files

abocquierCC commented on tests/query-editor.spec.ts:
rename this tests file according to the scenario and place in a director that represent the scope of the test

zahed942001 replied:
Done:
Renamed and seperated to multiple test files

abocquierCC commented on tests/constant-editor.spec.ts:
```diff
@@ -1,9 +1,31 @@
     const now = new Date().toISOString().replace('T', ' ').replace('Z', '');
     console.log(`[${now}] ${message}`);
 }
+async function fillPairAndClickAdd({nameInput, valueInput, name, value, addButton, label, page}: { nameInput: Locator, valueInput: Locator, name: string, value: string, addButton?: Locator, label: string, page: Page }) {
```
Use html components' ids, add some if their are not defined

zahed942001 replied:
Done, 
Created Ids and used them in my test to reference components

abocquierCC commented on .config/Dockerfile:
```diff
@@ -1,4 +1,4 @@
-ARG grafana_version=latest
+ARG grafana_version=9.5.21
```
remove if not necessary

zahed942001 replied:
Done:
Removed

abocquierCC commented on .config/docker-compose-base.yaml:
```diff
@@ -7,7 +7,7 @@ services:
       args:
         grafana_image: ${GRAFANA_IMAGE:-grafana-enterprise}
-        grafana_version: ${GRAFANA_VERSION:-11.3.2}
+        grafana_version: ${GRAFANA_VERSION:-9.5.21}
```
remove if not necessary

zahed942001 replied:
Done:
switched back

abocquierCC commented on .github/workflows/bundle-stats.yml:
```diff
@@ -4,6 +4,7 @@ on:
   pull_request:
     branches:
       - main
```
remove this before merging

zahed942001 replied:
The entire file or just the first line?

abocquierCC replied:
I think you should change dev to allow all branches starting the ci

abocquierCC commented on .github/workflows/ci.yml:
```diff
@@ -5,10 +5,12 @@ on:
     branches:
       - master
       - main
+      - dev
```
remove this before merging or allow all branches beginning

abocquierCC replied:
outdated and changed

abocquierCC commented on .github/workflows/ci.yml:
```diff
@@ -5,10 +5,12 @@ on:
     branches:
       - master
       - main
+      - dev
```
remove this before merging or allow all branches beginning

abocquierCC replied:
outdated and changed

abocquierCC commented on .github/workflows/ci.yml:
```diff
@@ -135,22 +137,29 @@ jobs:
+        id: filter
+        run: |
+          echo '${{ steps.resolve.outputs.matrix }}' | python3 scripts/filter_grafana_versions.py > filtered.json
+          echo "matrix=$(cat filtered.json)" >> $GITHUB_OUTPUT
```
a cleaner and easier solution is to use the filter input of `e2e-version` action

abocquierCC commented on .github/workflows/ci.yml:
```diff
@@ -204,7 +213,7 @@ jobs:
         if: ${{ always() && steps.run-tests.outcome == 'failure' }}
         with:
-          name: ${{ matrix.GRAFANA_IMAGE.NAME }}-v${{ matrix.GRAFANA_IMAGE.VERSION }}-${{github.run_id}}-server-log
+          name: ${{ matrix.GRAFANA_IMAGE.name }}-v${{ matrix.GRAFANA_IMAGE.version }}-${{github.run_id}}-server-log
```
any reason for changing uppercase props to lowercase ?

abocquierCC replied:
no reason ?

abocquierCC commented on docker-compose.yaml:
```diff
@@ -1,14 +1,20 @@
+      interval: 10s
+      timeout: 5s
+      retries: 6
+      start_period: 10s
```
I thought we didn't need to modify this Dockerfile too ?

abocquierCC replied:
outdated and changed

abocquierCC commented on playwright.config.ts:
```diff
@@ -4,48 +4,31 @@ import { dirname } from 'node:path';
+      testDir: './tests',
+      testMatch: ['*.spec.ts'],
+      use: {
+        ...devices['Desktop Chrome'],
```
allow other main devices

zahed942001 replied:
Done:
Added Desktop Firefox/Safari

abocquierCC replied:
(I don't see Safari)

zahed942001 replied:
WebKit (used for Safari-like testing) is the least supported in real-world dashboards and admin panels like Grafana.
it's unstable in CI and needs extra configuration, it has alot of console warnings and the behavior of the dashboard is so buggy, and it's stability is so poor in playwright test, sometimes it works, sometimes it doesn't (for the same code)
that is why it's not added to our CI anymore

abocquierCC replied:
I agree to support Chrome and Firefox only

abocquierCC commented on provisioning/datasources/datasources.yml:
```diff
@@ -2,11 +2,11 @@ apiVersion: 1
     jsonData:
-      path: http://host.docker.internal:8080
+      path: http://warp10:8080
       access: PROXY
```
I still not sure of this

(remove these change and create your own provisining for test)

zahed942001 replied:
Done:
Created new provisioning for my own tests

abocquierCC commented on scripts/filter_grafana_versions.py:
you can remove this (not used in CI with previous comments)

zahed942001 replied:
Done:
switched by grafana variable named skip-grafana-dev-image asigned to true

abocquierCC commented on tests/config/docker-compose-plugin.yaml:
```diff
@@ -0,0 +1,44 @@
+    build:
+      context: ../../.config
+      args:
+        grafana_version: ${GRAFANA_VERSION:-9.5.21}
```
just update to the most recent version we use

zahed942001 replied:
Done:
Version updated to 11.5.4

abocquierCC commented on tests/plugin_test.spec.ts:
Add context to your test file: correct dir name > correct filename > a comment at the top of the file explaining briefly what is tested and what we need to know about

zahed942001 replied:
Done:
Added tests descriptions

abocquierCC reviewed (commented):
Minor changes needed about the configuration. Minor except for the python script that filter version)

I'll wait for the other tests :)

abcqr commented on tests/requests/variables/const_variable_test.spec.ts:
```diff
@@ -21,7 +27,7 @@ test('Single constant substitution (Datasource constant)', async ({ page }) => {
     const query = `$myConstVar TOLONG 20 +`;
     log('--> Sending final test query');
-    const { payload, response, responseBody } = await executeQuery(page, dsName, query, constValue);
+    const { payload, response, responseBody } = await executeQuery_2(page, dsName, query, constValue);
```
executeQuery_2 doesn't seem like a good name
What is different from the first one ?

abcqr commented on tests/requests/variables/interval_variable_test.spec.ts:
```diff
@@ -0,0 +1,41 @@
+    const { payload } = await executeQuery_2(page, dsName, query);
+
+    // Assert variable name is in the query (the actual value will depend on Grafana's selection)
+    expect(payload.queries[0].expr).toContain('$myInterval');
```
If I understand well:
- the query in the query editor should be something simple like '1 2 +' the result attempt should be '3'
- what you should expect is not only $myInterval but the full string compute by the plugin:
      - `''" 'myConstVar' STORE`
      - `[ '' ] 'myConstVar_list' STORE`
      
(Same for other variable tests)

abcqr reviewed (commented):
Good ! Some change to make to test more precisely variable features

zahed942001 committed:
config(playwright): add environment configuration for tests

- update Playwright configuration file
- modify ConfigEditor for compatibility with our tests
- add new libraries required for end-to-end testing
- update Grafana version in Dockerfiles

zahed942001 committed:
add(environment): add environment configuration and provisioning for tests

- add provisioning environment for end-to-end tests
- add Dockerfile
- add Warp10 configuration file

zahed942001 committed:
add(tests): add e2e tests

- add tests by category (see doc/plugin-e2e-test-summary.md for details)

zahed942001 committed:
add(workflows): add CI workflows for e2e tests

- set up GitHub Actions for Playwright e2e testing

zahed942001 committed:
build(dist): update build files

zahed942001 committed:
docs(e2e): add documentation for e2e tests

zahed942001 committed:
fix(workflow)

zahed942001 commented:
Add Playwright E2E Tests, CI Workflows, and Documentation

- config(playwright): add environment configuration for tests
- add(environment): add environment configuration and provisioning for tests
- add(tests): add e2e tests
- add(workflows): add CI workflows for e2e tests
- build(dist): update build files
- docs(e2e): add documentation for e2e tests
- fix(workflow)

zahed942001 committed:
fix(ci.yml): remove docker-compose and change configuration

zahed942001 committed:
fix(test): minor change on repo name

abocquierCC commented:
merged
